cli/command/volume/create.go

```diff
@@ -33,24 +35,29 @@ func newCreateCommand(storageosCli *command.StorageOSCli) *cobra.Command {
 		Short: "Create a volume",
 		Args:  cli.RequiresMaxArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			if len(args) == 1 {
-				if opt.name != "" {
-					fmt.Fprint(storageosCli.Err(), "Conflicting options: either specify --name or provide positional arg, not both\n")
-					return cli.StatusError{StatusCode: 1}
-				}
-				opt.name = args[0]
+			var posarg string
+			if len(args) > 0 {
+				posarg = args[0]
 			}
+
+			var err error
+			opt.namespace, opt.name, err = parseNamespaceVolume(opt.namespace, opt.name, posarg)
+			if err != nil {
+				return err
+			}
+
 			return runCreate(storageosCli, opt)
 		},
 	}
+
 	flags := cmd.Flags()
 	flags.StringVar(&opt.name, "name", "", "Volume name")
 	flags.Lookup("name").Hidden = true
 	flags.StringVarP(&opt.description, "description", "d", "", "Volume description")
 	flags.IntVarP(&opt.size, "size", "s", 5, "Volume size in GB")
 	flags.StringVarP(&opt.pool, "pool", "p", "default", "Volume capacity pool")
 	flags.StringVarP(&opt.fsType, "fstype", "f", "", "Requested filesystem type")
-	flags.StringVarP(&opt.namespace, "namespace", "n", "default", "Volume namespace")
+	flags.StringVarP(&opt.namespace, "namespace", "n", "", "Volume namespace")
```

Comment:
cobra will show the default value in brackets on the CLI help.  We should still show the default in the say way as the other parameters, even if Cobra is not responsible for setting it.  Does the format work if you just append "
 (default)" to the helptext?

Reply:
Thanks. I have made this change, using the append to help-text method.